.github/workflows/check-labels.yml

```diff
@@ -8,16 +8,39 @@ jobs:
   check-labels:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout sources
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-          ref: ${{ github.event.pull_request.head.ref }}
-          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Pull image
+        env:
+          IMAGE: paritytech/ruled_labels:0.3.2
+        run: docker pull $IMAGE
+
       - name: Check labels
-        run: bash ${{ github.workspace }}/scripts/ci/github/check_labels.sh
         env:
+          IMAGE: paritytech/ruled_labels:0.3.2
+          MOUNT: /work
           GITHUB_PR: ${{ github.event.pull_request.number }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
-          BASE_SHA: ${{ github.event.pull_request.base.sha }}
+          API_BASE: https://api.github.com/repos
+          REPO: ${{ github.repository }}
+          RULES_PATH: labels/ruled_labels
+          CHECK_SPECS: specs_polkadot.yaml
+        run: |
+          echo "REPO: ${REPO}"
+          echo "GITHUB_PR: ${GITHUB_PR}"
+          # Clone repo with labels specs
+          git clone https://github.com/paritytech/labels
+          # Fetch the labels for the PR under test
+          labels=$( curl -H "Authorization: token ${GITHUB_TOKEN}" -s "$API_BASE/${REPO}/pulls/${GITHUB_PR}" | jq '.labels | .[] | .name' | tr "\n" ",")
+
+          if [ -z "${labels}" ]; then
+            echo "No labels found. Please add labels A, B and C labels"
+            exit 1
+          fi
```

Comment:
Same comment than above, but here it would make even more sense to let the rules do their work and not make any assumptions about those A, B and C labels.

Reply:
fixed